### ISSUE TITLE
What to do after use Inception_resnet_v1

### ISSUE THREAD

**Issue by BingoTiger** [davidsandberg/facenet#77]
These are some codes in facnet_train_classifier.py, from line 98 to line 106. I don't understand what the 'w' and 'b' mean here. In my understanding, 'n' is the number of cell of the full connection layer, 'm' is the number of classes. Is this wrong? And what's the meaning of 'w' and 'b'?

# Build the inference graph
        prelogits, _ = network.inference(image_batch, args.keep_probability, 
            phase_train=True, weight_decay=args.weight_decay)
        with tf.variable_scope('Logits'):
            n = int(prelogits.get_shape()[1])
            m = nrof_classes
            w = tf.get_variable('w', shape=[n,m], dtype=tf.float32, 
                initializer=tf.truncated_normal_initializer(stddev=0.1), 
                regularizer=slim.l2_regularizer(args.weight_decay),
                trainable=True)
            b = tf.get_variable('b', [m], initializer=None, trainable=True)
            logits = tf.matmul(prelogits, w) + b

Thanks.

**Comment by davidsandberg** [comment]
Yes, you are right. This is just the final fully connected layer, where w is weights and b is biases. It was broken out from the inception-resnet-v1 model since it depends a lot on what you want to do etc.
But I think this could be made a lot cleaner by just using the slim.fully_connected layer instead.

**Comment by BingoTiger** [comment]
There is still one thing I don't understand. In the final fully connected layer, there are 1792 cells, why the number of w is 1792*6 or the number of b is 6? In my mind, each cell should have one weight and one bias so the number of b and w should both be 1792.

**Comment by davidsandberg** [comment]
@BingoTiger Could it be that the batch size is 6 in the case that you are looking at?

**Comment by BingoTiger** [comment]
Sorry, I didn't describe clearly. Batch size is 10, but m is 6. I use a small sample of 6 people, there are 6 subfolders after align. Is there any relation? I think m is the number of classes, so there is no problem that m is 6, but why b use a parameter as [m]?
Thanks.

**Comment by davidsandberg** [comment]
Ok, now I think I understand how you mean.
Above you say that "In my mind, each cell should have one weight and one bias so the number of b and w should both be 1792", but this is not correct. In this case the number of biases should be equal to the number of classes.
https://en.wikipedia.org/wiki/Matrix_multiplication
Below are the dimensions of the layer, where n = 1792 and m = 6. Since it's a classifier we want to number of outputs to be equal to the number of classes (m).
```
prelogits *    w     +    b
(1 x n)   * (n x m)  + (1 x m) => (1 x m)
```

**Comment by BingoTiger** [comment]
Does the 'logits' mean the result probabilities after softmax?
I'm sorry for my poor English.
Thank you very much.

**Comment by davidsandberg** [comment]
Yes, that is correct.
